transform-plugins/src/main/java/io/cdap/plugin/Compressor.java

```diff
@@ -66,6 +69,26 @@ public Compressor(Config config) {
     this.config = config;
   }
 
+  @Override
+  public void prepareRun(StageSubmitterContext context) throws Exception {
+    super.prepareRun(context);
+    parseConfiguration(config.compressor, context.getFailureCollector());
+
+    // Initialize the required member maps and then: if a field is in input and output and comp, then
+    //    set it to transform; if it's in input and output but not comp, ignore it; if it's in input and
+    //    not output, drop it.
+    List<String> inFields = TransformLineageRecorderUtils.getFields(context.getInputSchema());
+    List<String> outFields = TransformLineageRecorderUtils.getFields(context.getOutputSchema());
+    List<String> idFields = outFields.stream()
+      .filter(field -> !compMap.containsKey(field) || compMap.get(field) == CompressorType.NONE)
+      .collect(Collectors.toList());
+
+    context.record(TransformLineageRecorderUtils.eachInToSomeOut(inFields, outFields, idFields,
+      "compress", "Used the specified algorithm to compress the field.",
+      "drop", "Dropped fields not included in the output.",
+      "identity", "Copied values of fields not marked for operation."));
```

Comment:
This description is too specific for user, just say `Unchanged field in the transform`, same for other places